r/src/compute.cpp

```diff
@@ -316,6 +316,19 @@ std::shared_ptr<arrow::compute::FunctionOptions> make_compute_options(
     return std::make_shared<Options>(max_splits, reverse);
   }
 
+  if (func_name == "utf8_slice_codeunits") {
+    using Options = arrow::compute::SliceOptions;
+    int64_t start = 1;
+    int64_t stop = -1;
```

Comment:
I think this still may need updating to reflect the default value of stop supplied by C++

Reply:
See the final comment here for more info but happy to explain if that doesn't make sense: https://issues.apache.org/jira/browse/ARROW-13259

Reply:
yes !!!! 
at the moment, I have added the additional "+1" to the R functions
if the backend changes, Iĺl adapt the functions

Reply:
ready !! i added new tests and I commented the expected results

Reply:
Sorry, I didn't explain this properly, my fault!  What I mean is that here stop has been set to `-1` but the default C++ value isn't `-1`, so the default value here probably shouldn't be `-1` either.

Check out this line of code here that shows the default value of `stop` in the C++: https://github.com/apache/arrow/blob/7114c4b6fdb63 9b3500d77cfd66649af8c5c5e6b/cpp/src/arrow/compute/api_scalar.h#L205-L206 

The default value of `stop` is `std::numeric_limits<int64_t>::max()` which is the biggest int64.  So if you don't supply a value for `stop` and just use the default, this allows you to slice to the end of the string.

In some of the other functions in this file, this line has been used to set the default values to the ones from the C++ code:
`std::make_shared<Options>(Options::Defaults());`

Maybe instead of manually setting the value of stop to `-1`, if you use the line above, it might make it easier to fix some of the failing tests as now you'll be able to slice to the end of a string.  If this doesn't make sense, let me know!